cpp/include/Ice/Comparable.h

```diff
@@ -6,8 +6,6 @@
 #define ICE_COMPARABLE_H
 
 #include <functional>
-#include <memory>
```

Comment:
This change reverts back this file to the way it was before the new proxy mapping. All the targetXxx functions are used only shared_ptr now, even though they also "work" with optional.  I changed these functions temporarily to detect all uses with optional.